gateway/ln-gateway/src/ng/register.rs

```diff
@@ -0,0 +1,149 @@
+use std::time::Duration;
+
+use fedimint_client::sm::{OperationId, State, StateTransition};
+use fedimint_client::DynGlobalClientContext;
+use fedimint_core::encoding::{Decodable, Encodable};
+use fedimint_core::task::sleep;
+use fedimint_ln_common::api::LnFederationApi;
+use fedimint_ln_common::LightningGateway;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use tracing::error;
+use url::Url;
+
+use super::GatewayClientContext;
+
+#[cfg_attr(doc, aquamarine::aquamarine)]
+/// State machine that registers the lightning gateway with
+/// the federation.
+///
+/// ```mermaid
+/// graph LR
+/// classDef virtual fill:#fff,stroke-dasharray: 5 5
+///
+///    Register -- register lightning gateway succeeded --> WaitForTTL
+///    Register -- register lightning gateway failed --> Register
+///    WaitForTTL -- wait for time to live to expire --> Register
+/// ```
+#[derive(Debug, Clone, Eq, PartialEq, Decodable, Encodable)]
+pub enum GatewayRegisterStates {
```

Comment:
IMO, `RegisterFederation` and `RegisterFederationStates` might be more inntuitive names for this SM